### PR TITLE
Base: merge two identical CPP cases

### PR DESCRIPTION
**Pull request by compnerd**
The two cases are identical.  Merge them to avoid the unnecessary
duplication.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @millenomi

File: CoreFoundation/Base.subproj/CFInternal.h [hunk 1]
```diff
@@ -504,37 +504,18 @@ CF_EXPORT id __NSDictionary0__;
...
-    (void)pthread_mutex_lock(LP); })
-
-#define __CFUnlock(LP) ({ \
-    (void)pthread_mutex_unlock(LP); })
-
-#define __CFLockTry(LP) ({ \
-    pthread_mutex_trylock(LP) == 0; })
-
-#elif DEPLOYMENT_TARGET_EMBEDDED || DEPLOYMENT_TARGET_EMBEDDED_MINI
+#if DEPLOYMENT_TARGET_MACOSX || DEPLOYMENT_TARGET_EMBEDDED || DEPLOYMENT_TARGET_EMBEDDED_MINI
```
Comment (millenomi) [inline CoreFoundation/Base.subproj/CFInternal.h]: Can you make this line just read: `#if TARGET_OS_MAC`?

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**